play/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/CommitEdit.kt

```diff
@@ -15,13 +15,16 @@ import javax.inject.Inject
 @Suppress("LeakingThis")
 @DisableCachingByDefault
 internal abstract class CommitEdit @Inject constructor(
-        private val gradle: Gradle,
         extension: PlayPublisherExtension,
         private val executor: WorkerExecutor,
 ) : PublishTaskBase(extension) {
     init {
+        val buildFailedProvider = project.provider {
+            project.gradle.taskGraph.allTasks.any { it.state.failure != null }
+        }
```

Comment:
This is not working to me @ZacSweers , based on my tests, the `[onlyIf` closure is evaluated at configuration time](https://gradle-community.slack.com/archives/C013WEPGQF9/p1683047848634369?thread_ts=1682304058.131779&cid=C013WEPGQF9) when it shouldn't be and of course the task outcome is not yet populated.

Reply:
@SUPERCILEX  how do you feel about making `CommitEdit` task a "success only" one and moving the `apiService.cleanup` logic to the `PlayApiService` by adding `AutoCloseable`?

From the [docs](https://docs.gradle.org/current/userguide/build_services.html):
> A build service implementation can also optionally implement AutoCloseable, in which case Gradle will call the build service instance’s close() method when it discards the service instance. This happens some time between completion of the last task that uses the build service and the end of the build.

I could let commit run only if all tasks calling it with `finalizedBy` also does `dependsOn` in the opposite way, as suggested [here](https://gradle-community.slack.com/archives/C013WEPGQF9/p1683045329792689?thread_ts=1682304058.131779&cid=C013WEPGQF9).

Later `PlayApiService` will be responsible of cancelling any pending transaction on its `close` method.

Reply:
I think I had considered using AutoCloseable before but to handle both success and failure. I think using it just for failure cleanup makes sense, though I'm not clear on how you prevent the CommitEdit task from trying to commit if a task failed (can you paste whatever is in the slack?).

Reply:
Sure, it's basically linked them both ways:
```kotlin
publishApkTasl.configure { finalizedBy(commitTask) }
commitTask.configure { dependsOn(publishApkTasl) }
```

But now I'm seeing another issue, we have 4 potential (if I remember correctly) tasks `finalizedBy`. We want `commit` to run if ANY of those has succeed, not to force all of them to run 🤔

Reply:
I managed to get it to work by introducing `AutoCloseable` to `cleanup` and `OperationCompletionListener` to keep track of failed tasks on its new `tasksHasFailed` property.
Also added the `usesService` API call allowing the `PlayApiService` to close right after the last task using it is done.

The behavior seems to be the same, and tests are passing now (expect for one which I commented).

Can you review this now? thanks!